src/http/cookie.cr

```diff
@@ -42,9 +47,29 @@ module HTTP
       "#{@name}=#{URI.escape value}"
     end
 
+    # Returns the `Time` at which this cookie will expire, or `nil` if it will not expire.
+    # Uses *max-age* and *expires* values to calculate the time.
+    def expiration_time
+      max_age = @max_age
+      expires = @expires
+      if max_age
+        creation_time + max_age
+      elsif expires
+        expires
+      else
+        nil
+      end
+    end
+
     def expired?
-      if e = expires
-        e < Time.now
+      max_age = @max_age
```

Comment:
This method has duplicated code from `expiration_time` and should use that method instead. This way it becomes very simple:

```
@max_age == 0.seconds || expiration_time.try &.<(Time.now) || false
```